### DISCUSSION TITLE
"Package resolution errors must be fixed before building"

### DISCUSSION THREAD

**Discussion by sherryycxie** [CS342/.github#6] [Technical Questions] [answered]
### In what area do you have a technical challenge?

SwiftUI

### Description

I just created my own branch of the CS342 Application and am trying to build it. However, I could not do so because Xcode tells me I am missing the package products for Paul and Oliver: 
<img width="253" alt="Screenshot 2023-01-19 at 11 34 13 PM" src="https://user-images.githubusercontent.com/85938088/213641539-8a3c9960-1ad1-4404-9536-6428e8b81725.png">

How should I fix this issue?

### Reproduction

I just clicked "build" in Xcode and reached this stage.

### Expected behavior

I expect that these packages should be there and I should be able to build the application.

### Code of Conduct

- [X] I agree to follow this course's Code of Conduct and Contributing Guidelines

**Comment by vishnuravi** [comment]
Hi @sherryycxie, in Xcode click `File` > `Packages` > `Reset Package Caches`. This might solve your issue. Let us know if it does.

**Comment by PSchmiedmayer** [comment, marked as answer]
@sherryycxie Thank you for posting your question!

As @vishnuravi noted, sometimes Xcode has a few problems cleaning out issues or invalidating the dependency check. If you run into problems that you can't easily sport the error for, we do generally advise:

- Check the [report navigator for issues](https://help.apple.com/xcode/mac/11.4/index.html?localePath=en.lproj#/dev21d56ecd4) (especially also on a Project-based level regarding your Swift Package)
- Clean all issues using **Product > Clear All issues** and try to run the build again: **Product > Build (⌘B)**
- To rebuild all files in a project, choose **Product > Clean Build Folder (⇧⌘K)** and try to run the build again: **Product > Build (⌘B)**

If you modify your Swift Package and add a target, make sure that all targets are separated by a comma (the list of targets, dependencies, and products in a Swift Package Manifest are just arrays where you have to separate elements by commas). If you add a new target to your Swift Package, link it to your application as detailed in [Organizing your code with local packages](https://developer.apple.com/documentation/xcode/organizing-your-code-with-local-packages).

Can you share your Swift Package Manifest file with us in a comment? GitHub allows you to share code snippets including syntax highlighting in comments and issues: [Creating Code Blocks - Syntax highlighting](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting).

Also be sure that you have at least one Swift file in each of your target folders.

**Reply by PSchmiedmayer** [reply]
@sherryycxie Feel free to let us know if any of the suggestions solved your problem.
If so, please answer the corresponding comment as an answer so we can mark the discussion as solved and others easily know where to find the answer 👍

**Reply by sherryycxie** [reply]
Yep! I believe my problem is solved. Thank you so much!
